sdk/core/azure-core-amqp/inc/azure/core/amqp/connection.hpp

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-Licence-Identifier: MIT
+
+#pragma once
+
+#include "common/async_operation_queue.hpp"
+#include "connection_string_credential.hpp"
+#include "models/amqp_value.hpp"
+#include "session.hpp"
+#include <azure/core/credentials/credentials.hpp>
+#include <chrono>
+#include <memory>
+#include <string>
```

Comment:
formatting nit: add a new line between groups of includes, split between `"` surrounded ones, `<azure>` ones, and `non-azure` ones (i.e. STL). It helps clang-format order them consistently.

Like this:
https://github.com/Azure/azure-sdk-for-cpp/blob/d693099d44c4f965cde6ff06d6f3cd0358ef01a8/sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_client.hpp#L6-L15

Reply:
Our clang-format specifies that it uses the LLVM formatting style, which is defined to be the following [header grouping](https://llvm.org/docs/CodingStandards.html#include-style). If we want to use a different header grouping, we can change the clang-format to include our desired ordering.

We are currently using the [IncludeBlocks](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#includeblocks) `Preserve` option to achieve a manual ordering. IMHO we should only override the clang-format rules if it is required for correctness.

If we want to have the ordering be global ones, azure ones, and system headers, we can do that using the [IncludeBlocks](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#includeblocks) option.